### PR TITLE
defer closing the ReadCloser in ObjectStoreGRPCServer.GetObject

### LINKED ISSUE #1093: Need to defer closing the the ReadCloser in ObjectStoreGRPCServer.GetObject
According to https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/making-requests.html, under "Handling Operation Response Body", if we call an operation that returns a `Body` in the response object, we need to `Close()` it when we're done with it. The same is true for GCP and Azure.

We should add the `defer` here: https://github.com/heptio/ark/blob/09042a9015e17d8dc00399a0c516a604d5f94f9d/pkg/plugin/object_store.go#L287

### PR DESCRIPTION
**Pull request by DheerajSShetty**
Signed-off-by: DheerajSShetty <dheerajs@vmware.com>

Fixes #1093

### CODE REVIEW

File: changelogs/unreleased/1093-DheerajSShetty [hunk 1]
```diff
@@ -0,0 +1 @@
+Need to defer closing the the ReadCloser in ObjectStoreGRPCServer.GetObject
```
Comment (nrb) [inline changelogs/unreleased/1093-DheerajSShetty]: Could you rename this file to `1236-DheerajSShetty` please? Our current format is to use the PR number.

Comment (DheerajSShetty) [reply changelogs/unreleased/1093-DheerajSShetty]: Ok

**Comment by skriss** [comment]
LGTM @DheerajSShetty. Were you able to test this? If not, maybe @nrb or @carlisia can help you out tomorrow. Happy to ship this in v0.11 as long as there's been some basic validation on it.

**Comment by DheerajSShetty** [comment]
Created a back up, deleted the namespace and called restore back up to test this change.

**Comment by skriss** [comment]
@DheerajSShetty LGTM - please squash it into a single commit

**Comment by skriss** [comment]
@nrb or @carlisia pls review + merge

**Pull request merged**